File: CS/DXSample/ViewModel/HomeViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace DXSample.ViewModel {
    public class HomeViewModel {
        private INavigationService NavigationService { get { return this.GetService<INavigationService>(); } }

        public HomeViewModel() {  }

        public void NavigateDetails() {
            NavigationService.Navigate("DetailView", null, this);
        }
    }
}

File: VB/DXSample/ViewModel/HomeViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace DXSample.ViewModel
	Public Class HomeViewModel
		Private ReadOnly Property NavigationService() As INavigationService
			Get
				Return Me.GetService(Of INavigationService)()
			End Get
		End Property

		Public Sub New()
		End Sub

		Public Sub NavigateDetails()
			NavigationService.Navigate("DetailView", Nothing, Me)
		End Sub
	End Class
End Namespace
